src/Argument.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Typhoon\Type;
+
+/**
+ * @api
+ * @psalm-immutable
+ */
+final class Argument
```

Comment:
I named this argument instead of property, because we check type of the actual passed argument, not the declared property (it's type we already know).